src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,61 @@
 'use strict';
 
+'use strict';
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
+
+  isAlive() {
+    return this.health > 0;
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (!this.isAlive()) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
```

Comment:
[CODE PERFORMANCE] Instead of using `indexOf` and `splice` to remove the dead animal from `Animal.alive`, consider using `filter`. This avoids the nested loop and is more efficient and cleaner.

Reply:
ok